### PR TITLE
Update plugin names and descriptions to better describe their features

### LINKED ISSUE #1046: Review current plugin names and their descriptions for their accuracy and helpfulness
See https://github.com/WordPress/performance/issues/1031#issuecomment-1989043559: We should revisit our current standalone plugins' names and descriptions for how accurate and useful they are. In particular, we should consider the following:
* Plugin names and descriptions are used to describe features within Performance Lab.
* Our plugins are not _products_. They are features for WordPress core. As such, we should avoid plugin names that sound too much like a product / brand name, and instead simply find a good name that clarifies the feature. Example: "Optimization Detective" doesn't clarify what it does.
* Additionally, we should avoid technicalities in the name, to make them intuitive. Example: "WebP Uploads" would probably be better as something like "WebP Images" (the term "Uploads" is only used because of Media Library jargon). "Speculation Rules" should probably be something like "Speculative preloading" because the Speculation Rules API is a technical detail.
* Plugin descriptions should not mention the word "plugin". They can just start with a third-person verb (e.g. "Adds xyz" or similar).

For reference, here are our current plugin names and descriptions to review:
* Auto-sizes for Lazy-loaded Images
    * This plugin implements the HTML spec for adding `sizes="auto"` to lazy-loaded images.
* Dominant Color Images
    * Adds support to store the dominant color of newly uploaded images and create a placeholder background of that color.
* Optimization Detective
    * Improves accuracy of optimizing the loading of the LCP image by leveraging client-side detection with real user metrics. Also enables output buffering of template rendering which can be filtered.
* Performant Translations
    * Making internationalization/localization in WordPress faster than ever before.
* Speculation Rules
    * Uses the Speculation Rules API to prerender linked URLs upon hover by default.
* WebP Uploads
    * Creates WebP versions for new JPEG image uploads if supported by the server.

Let's brainstorm.

cc @adamsilverstein @joemcgill @mukeshpanchal27 @swissspidy @westonruter

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Fixes #1046 (note that the prefixing discussion and implementation should be part of a separate issue)

This is a sub PR of #1100. Please merge that one first.

## Relevant technical choices

* Applies plugin names and descriptions from https://github.com/WordPress/performance/issues/1046#issuecomment-2005134234.
* Bumps minor version as needed since the name change is more significant than a patch version increase IMO.
* Adds changelog entries about the name change.

### CODE REVIEW

File: plugins/dominant-color-images/readme.txt [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-Stable tag:        1.0.2
+Stable tag:        1.1.0
 License:           GPLv2 or later
 License URI:       https://www.gnu.org/licenses/gpl-2.0.html
 Tags:              performance, images, dominant color
 
-Adds support to store the dominant color of newly uploaded images and create a placeholder background of that color.
+Displays placeholders based on an image's dominant color while the image is loading.
 
 == Description ==
```
Comment (westonruter) [inline plugins/dominant-color-images/readme.txt:14]: Perhaps the first line of the description should say:

> _This plugin was formerly known as Dominant Color Images._

File: plugins/speculation-rules/readme.txt [hunk 2]
```diff
@@ -9,15 +9,15 @@ License:           GPLv2 or later
 License URI:       https://www.gnu.org/licenses/gpl-2.0.html
 Tags:              performance, javascript, speculation rules, prerender, prefetch
 
-Uses the Speculation Rules API to prerender linked URLs upon hover by default.
+Enables browsers to speculatively prerender or prefetch pages when hovering over links.
 
 == Description ==
```
Comment (westonruter) [inline plugins/speculation-rules/readme.txt:14]: Ditto above about mentioning the old name.

File: plugins/webp-uploads/readme.txt [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-Stable tag:        1.0.6
+Stable tag:        1.1.0
 License:           GPLv2 or later
 License URI:       https://www.gnu.org/licenses/gpl-2.0.html
 Tags:              performance, images, webp
 
-Creates WebP versions for new JPEG image uploads if supported by the server.
+Converts images to more modern formats such as WebP or AVIF during upload.
 
 == Description ==
```
Comment (westonruter) [inline plugins/webp-uploads/readme.txt:14]: Ditto above about mentioning the old name.

File: plugins/dominant-color-images/readme.txt (hunk 1 above)
Comment (felixarntz) [reply plugins/dominant-color-images/readme.txt:14]: @westonruter How about putting those explanations at the end of the respective `== Description ==` section?

Comment (westonruter) [reply plugins/dominant-color-images/readme.txt:14]: Sounds good

Comment (felixarntz) [reply plugins/dominant-color-images/readme.txt:14]: Done in https://github.com/WordPress/performance/pull/1101/commits/0c04a92f8cdf1eb4067a0ca11f2c0b92ee4a7b24

File: plugins/speculation-rules/readme.txt (hunk 2 above)
Comment (felixarntz) [reply plugins/speculation-rules/readme.txt:14]: Done in https://github.com/WordPress/performance/pull/1101/commits/0c04a92f8cdf1eb4067a0ca11f2c0b92ee4a7b24

File: plugins/webp-uploads/readme.txt (hunk 3 above)
Comment (felixarntz) [reply plugins/webp-uploads/readme.txt:14]: Done in https://github.com/WordPress/performance/pull/1101/commits/0c04a92f8cdf1eb4067a0ca11f2c0b92ee4a7b24

**Review by westonruter — APPROVED** [review]

**Review by thelovekesh — APPROVED** [review]

**Pull request merged**